### PR TITLE
[ServiceBus] Improve AMQP Error handling

### PR DESCRIPTION
**Pull request by yunhaoling**
addressing issue:
https://github.com/Azure/azure-sdk-for-python/issues/15688
https://github.com/Azure/azure-sdk-for-python/issues/15749

- Operations failing due to `uamqp.errors.LinkForceDetach` caused by no activity on the connection for 10 minutes will now be retried internally.
- `uamqp.errors.AMQPConnectionError` errors with condition code `amqp:unknown-error` are now categorized into `ServiceBusConnectionError` instead of the general `ServiceBusError`.

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## 7.0.2 (Unreleased)
 
+**BugFixes**
+
+* Operations failing due to `uamqp.errors.LinkForceDetach` caused by no activity on the connection for 10 minutes will now be retried internally.
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:7]:
```suggestion
* Operations failing due to `uamqp.errors.LinkForceDetach` caused by no activity on the connection for 10 minutes will now be retried internally except for the session receiver case.
```

**Review by swathipil — APPROVED** [review]
lgtm (although I'm no expert)

File: sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py [hunk 2]
```diff
@@ -491,6 +493,7 @@ class AutoLockRenewTimeout(ServiceBusError):
     AMQPErrorCodes.UnauthorizedAccess: ServiceBusAuthorizationError,
     AMQPErrorCodes.NotImplemented: ServiceBusError,
     AMQPErrorCodes.NotAllowed: ServiceBusError,
+    AMQPErrorCodes.LinkDetachForced: ServiceBusConnectionError,
```
Comment (annatisch) [inline sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py:496]: What kind of error was this raised as before?

Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py:496]: it used to be the general `ServiceBusError` which would not be retried.

but after contemplating over it, I think this should be categorized into `ServiceBusConnectionError` and be retried.

**Review by rakshith91 — APPROVED** [review]

**Review by annatisch — APPROVED** [review]

**Pull request merged**